### ISSUE TITLE
Discussion: should we generate warning for unmatched `excluded` glob pattern?

### ISSUE THREAD

**Issue by GetToSet** [realm/SwiftLint#5078]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

I use `excluded` in `.swiftlint.yml` to prevent certain paths being linted.

```.swiftlint.yml
...

excluded:
  - "*/legacy/*.swift"
  - "./Scripts/**"
  - "./vendor/**"
  - "*/.build/**"
  - "./out/**"

...
```

SwiftLint currently generates a message starting with `warning: ` when there is no match for a specific glob pattern.

```bash
$ swiftlint lint
Linting Swift files at paths .
warning: Error parsing file system item: Error Domain=NSCocoaErrorDomain Code=260 "The file “vendor” couldn’t be opened because there is no such file." UserInfo={NSFilePath=vendor/, NSUnderlyingError=0x6000033f2370 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
warning: Error parsing file system item: Error Domain=NSCocoaErrorDomain Code=260 "The file “.build” couldn’t be opened because there is no such file." UserInfo={NSFilePath=*/.build/, NSUnderlyingError=0x6000033ec5d0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
warning: Error parsing file system item: Error Domain=NSCocoaErrorDomain Code=260 "The file “out” couldn’t be opened because there is no such file." UserInfo={NSFilePath=out/, NSUnderlyingError=0x6000033ec8a0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
...
```

For Xcode 15 beta, looks like those warnings are getting into the issues tab on the left sidebar, which are quite disturbing:

<img width="494" alt="Screenshot 2023-06-24 at 22 26 22" src="https://github.com/realm/SwiftLint/assets/8158163/a0540830-72fe-49a6-a67a-657d72ceb726">

Even though there're no matching directories, those glob entries were added intentially. For example, `*/.build/**` was added to `.gitignore` and to `excluded` section of `.swiftlint.yml` as well, ensuring that it won't cause any issues to SwiftLint if this directory gets accidenty created.

Should we keep generate warnings for unmatched `excluded` glob pattern, either with a more user-friendly warning message or remove it at all?

### Environment

* SwiftLint version: `0.52.3`.

* Installation method used: Homebrew.

* Which Xcode version are you using:

  ```
  Xcode 15.0
  Build version 15A5161b
  ```

**Comment by SimplyDanny** [comment]
Looks to me like tagging it with  `warning: ` is generally okay. I wonder, however, why the error is issued for excluded folders anyway.

**Comment by GetToSet** [comment]
> Looks to me like tagging it with `warning: ` is generally okay. I wonder, however, why the error is issued for excluded folders anyway.

These warning messages are actually a bit obscure. They only show up when there is no matching files or folders for each of the specified glob patterns.
